### PR TITLE
Fix kat_dns settings

### PR DESCRIPTION
**Pull request by originalsouth**
### Changes
In the current working main (1b389906bf79db4995296cc4f48155bc54ec3845), the settings for kat_dns are ignored. This PR addresses the issue.

### Issue link
N/A

### Demo
image
image

---

### Code Checklist
- [x] All the commits in this PR are properly PGP-signed and verified.
- [x] This PR only contains functionality relevant to the issue; tickets have been created for newly discovered issues.
- [ ] I have written unit tests for the changes or fixes I made.
- [x] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [x] I have performed a self-review of my code and refactored it to the best of my abilities.

### Communication
- [x] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [x] I have made corresponding changes to the documentation, if necessary.
- [x] I have included comments in the code to elaborate on what is not self-evident from the code itself, including references to issues and discussions online, or implicit behavior of an interface.

---
## Checklist for code reviewers:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---
## Checklist for QA:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

**Comment by originalsouth** [comment]
@underdarknl because of the current implementation of https://github.com/minvws/nl-kat-coordination/blob/788ffa2a19234612f6ec2772b41ec7b14ff2f0bc/boefjes/boefjes/plugins/kat_dns/main.py#L31-L72 we also yield a SOA record even of we only want an A record; I assume this is _not_ the desired behavior but double checking here.

**Comment by underdarknl** [comment]
Agree, ill fix.

**Comment by originalsouth** [comment]
Many thanks to @noamblitz for helping fix the tests!

**Comment by originalsouth** [comment]
Disregarding the SOA records, this branch is ready for merge.

**Comment by originalsouth** [comment]
@noamblitz according to @underdarknl you wrote this Boefje and hence we need your input. 

The changes allow a user to specify the record set. The question is whether we need the SOA record regardless of the specified records or not. The SOA, is the "authority" validating the record and hence for validation purposes might need to be present. What do you think?

**Comment by noamblitz** [comment]
I would not dare to imply that @Lisser his work is mine haha. However, I do not think we "need" the SOA record. Everything in KAT should work fine without it. If the user **wants** to leave out SOA records, they are free to do so I think.

File: boefjes/boefjes/plugins/kat_dns/schema.json [hunk 1]
```diff
@@ -10,10 +10,11 @@
     },
     "REMOTE_NS": {
       "title": "REMOTE_NS",
-      "maxLength": 45,
+      "maxLength": 15,
```
Comment (underdarknl) [inline boefjes/boefjes/plugins/kat_dns/schema.json:13]: I chose 45 to also allow for ipv6 nameserver ip's.

Comment (underdarknl) [inline boefjes/boefjes/plugins/kat_dns/schema.json:13]:
```suggestion
      "maxLength": 45, 
```

File: boefjes/boefjes/plugins/kat_dns/schema.json [hunk 2]
```diff
@@ -10,10 +10,11 @@
...
       "type": "string",
       "description": "The IP address of the DNS resolver you want to use.",
       "default": "1.1.1.1"
     }
-  }
+  },
+  "required": []
```
Comment (underdarknl) [inline boefjes/boefjes/plugins/kat_dns/schema.json:19]: This stanza should not be needed, did you see any errors that indicated it was in fact needed?

Comment (originalsouth) [reply boefjes/boefjes/plugins/kat_dns/schema.json:19]: Indeed, it is not needed, I wanted to make it conform to other Boefjes like:
boefjes/boefjes/plugins/kat_nmap_ip_range/schema.json
boefjes/boefjes/plugins/kat_dns/schema.json
boefjes/boefjes/plugins/kat_nmap_udp/schema.json
boefjes/boefjes/plugins/kat_masscan/schema.json
Perhaps they do not need it either?

File: boefjes/boefjes/plugins/kat_dns/schema.json (hunk 1 above)
Comment (originalsouth) [reply boefjes/boefjes/plugins/kat_dns/schema.json]: Addressed in 7020509b021935dcf87f20cc0fab976f4f69a37e.

Comment (originalsouth) [reply boefjes/boefjes/plugins/kat_dns/schema.json]: Addressed in 7020509b021935dcf87f20cc0fab976f4f69a37e.

File: boefjes/boefjes/plugins/kat_dns/schema.json (hunk 2 above)
Comment (originalsouth) [reply boefjes/boefjes/plugins/kat_dns/schema.json]: Addressed in 7020509b021935dcf87f20cc0fab976f4f69a37e.

**Comment by stephanie0x00** [comment]
## Checklist for QA:
- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
    - [x] I have managed to pass the onboarding flow
    - [x] Objects and Findings are created properly
    - [x] Tasks are created and completed properly
- [x] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.

### What works:
Onboarding looks good, parsing of DNS records seems to work as expected. The DNS records seem to match with a manual request of the DNS Records. 

### What doesn't work:
n/a

### Bug or feature?:
n/a

**Review by underdarknl — APPROVED** [review]

**Pull request merged**